src/Native/configure.cmake

```diff
@@ -226,6 +226,39 @@ check_include_files(
     linux/rtnetlink.h
     HAVE_LINUX_RTNETLINK_H)
 
+# getdomainname on OSX takes an 'int' instead of a 'size_t'
+# check if compiling with 'size_t' would cause a warning
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Weverything")
+check_cxx_source_compiles(
+    "
+    #include <unistd.h>
+	int main() { size_t namelen = 20; char name[namelen]; getdomainname(name, namelen); }
+    "
+    HAVE_GETDOMAINNAME_SIZET
+	FAIL_REGEX "-Wshorten"
+)
+set (CMAKE_REQUIRED_FLAGS ${PREVIOUS_CMAKE_REQUIRED_FLAGS})
```

Comment:
Thanks, @jasonwilliams200OK. I think this is a good idea in general. Unfortunately, I can't get it to work in this situation.
I tried the following:
```
check_prototype_definition(
    getdomainname
    "int getdomainname(char *name, size_t len)"
    "0"
    "unistd.h"
    HAVE_GETDOMAINNAME_SIZET)
```
But on my Ubuntu 14 box, that check always "Fails" with the following message in the CMakeError.log:
```
error: use of undeclared identifier 'getdomainname'
  cmakeRequireSymbol(0, &getdomainname);
```

From what I can discern, getdomainname is defined in /usr/include/unistd.h inside of `#if defined __USE_BSD || (defined __USE_XOPEN && !defined __USE_UNIX98)`.  Nothing is setting these defines in the check_prototype_definition call, thus it doesn't think this function exists. Therefore, the check fails and the HAVE_GETDOMAINNAME_SIZET variable never gets set. If you know how to make this work, I think it might be "niftier" as well, but I can seem to make it work.